cleanlab/filter.py

```diff
@@ -554,50 +582,90 @@ def reduce_prune_counts(prune_count_matrix, frac_noise=1.0):
+    """This is the simplest baseline approach. Just consider
+    anywhere argmax != labels as a label error.
 
     Parameters
     ----------
-    label_errors_bool : np.array (bool)
-      Contains True if the index of labels is an error, o.w. false
+    labels : np.array
+        A discrete vector of noisy labels, i.e. some labels may be erroneous.
 
     psx : np.array (shape (N, K))
-      P(s=k|x) is a matrix with K probabilities for all N examples x.
-      This is the probability distribution over all K classes, for each
-      example, regarding whether the example has label s==k P(s=k|x). psx
-      should computed using 3 (or higher) fold cross-validation.
+        P(label=k|x) is a matrix with K (noisy) probabilities for each of the
+        N examples x. This is the probability distribution over all K classes,
+        for each example, regarding whether the example has label labels==k P(labels=k|x).
+        psx should have been computed using 3 (or higher) fold cross-validation.
+
+    multi_label : bool
+        Set to True if labels is multi-label (list of lists, or np.array of np.array)
+
+    Returns
+    -------
+        A boolean mask that is true if the example belong
+        to that index is label error."""
+
+    assert (len(psx) == len(labels))
+    if multi_label:
+        # Todo: This needs to be tested.
+        return np.array(
+            [all(np.argsort(psx[i])[:len(j)] == sorted(j)) for i, j in enumerate(labels)])
+    return np.argmax(psx, axis=1) != np.asarray(labels)
+
+
+def find_label_issues_using_argmax_confusion_matrix(
+    labels,
+    psx,
+    calibrate=True,
+    filter_by='prune_by_noise_rate',
+):
+    """This is a baseline approach. That uses the confusion matrix
+    of argmax(psx) and labels as the confident joint and then uses cleanlab
+    (confident learning) to find the label issues using this matrix.
+
+    The only difference between this and find_label_issues is that it uses the confusion matrix
+    based on the argmax and given label instead of using the confident joint from
+    `count.compute_confident_joint`.
+
+    This method does not support multi-label labels.
+
+    Parameters
+    ----------
 
     labels : np.array
-      A binary vector of labels, which may contain label errors.
+        A discrete vector of noisy labels, i.e. some labels may be erroneous.
 
-    sorted_index_method : str ['normalized_margin', 'prob_given_label']
-      Method to order label error indices (instead of a bool mask), either:
-        'normalized_margin' := normalized margin (p(s = k) - max(p(s != k)))
-        'prob_given_label' := [psx[i][labels[i]] for i in label_errors_idx]
+    psx : np.array (shape (N, K))
+        P(label=k|x) is a matrix with K (noisy) probabilities for each of the
+        N examples x. This is the probability distribution over all K classes,
+        for each example, regarding whether the example has label labels==k P(labels=k|x).
```

Comment:
I'd replace this text with:
"Each row of this matrix corresponds to an example `x` and contains the model-estimated probabilities that `x` belongs to each possible class. The columns must be ordered such that these probabilities correspond to class 0,1,2,... Properly formatted `psx` may be obtained from the `predict_proba()` function in sklearn."

Reply:
made some adjustments, but added a few things. most important thing to include here is that these are computed via cross validation which isn't obvious with sklearn for any model, but took some thoughts from here and added everywhere

Reply:
```python
psx : np.array (shape (N, K))
        P(label=k|x) is a matrix with K model-predicted probabilities.
        Each row of this matrix corresponds to an example x and contains the model-predicted
        probabilities that x belongs to each possible class.
        The columns must be ordered such that these probabilities correspond to class 0,1,2,...
        `psx` should have been computed using 3 (or higher) fold cross-validation.
```